src/main/java/org/opensearch/knn/index/memory/NativeMemoryLoadStrategy.java

```diff
@@ -119,6 +123,55 @@ public NativeMemoryAllocation.IndexAllocation load(NativeMemoryEntryContext.Inde
             );
         }
 
+        @Override
+        public NativeMemoryAllocation.IndexAllocation load(NativeMemoryEntryContext.IndexEntryContext indexEntryContext)
+            throws IOException {
+            final Path absoluteIndexPath = Paths.get(indexEntryContext.getKey());
+            final KNNEngine knnEngine = KNNEngine.getEngineNameFromPath(absoluteIndexPath.toString());
+            if (knnEngine != KNNEngine.FAISS) {
+                // We will support other non-FAISS native engines (ex: NMSLIB) soon.
+                return loadWithAbsoluteIndexPath(indexEntryContext);
+            }
+
+            final FileWatcher fileWatcher = new FileWatcher(absoluteIndexPath);
+            fileWatcher.addListener(indexFileOnDeleteListener);
+            fileWatcher.init();
+
+            final Directory directory = indexEntryContext.getDirectory();
+
+            // Ex: Input -> /a/b/c/_0_NativeEngines990KnnVectorsFormat_0.vec
+            // Output -> _0_NativeEngines990KnnVectorsFormat_0.vec
+            final String logicalIndexPath = absoluteIndexPath.getFileName().toString();
+
+            final int indexSize = (int) directory.fileLength(logicalIndexPath);
+
+            try (IndexInput readStream = directory.openInput(logicalIndexPath, IOContext.READONCE)) {
```

Comment:
Please check when openInput is called what is the memory spike we are observing.

Reply:
Need more input for this, which issue are you referring to??

Reply:
I am not talking about any issue here. So my question is in your tests when you the queries are run, can we graph the spikes in memory during the query. As IndexInput.openInput maps the files to memory. So want to know if there is any spike we are seeing and how much is that spike actually.